### PR TITLE
[MM-31934] Global policy form

### PR DESCRIPTION
**Pull request by BenCookie95**
#### Summary
New global policy form. This PR contains code for the hybrid input as well (linked below)

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-31934

#### Related Pull Requests
https://github.com/mattermost/mattermost-server/pull/17308
https://github.com/mattermost/mattermost-webapp/pull/7726

#### Screenshots
Screen Shot 2021-04-01 at 4 14 37 PM

```release-note
This is a new page to manage the global retention policy of a mattermost instance. It's hidden behind a feature flag called CustomDataRetentionEnabled. 

The page contains 2 hybrid inputs to control the global settings for message and file retention.
```

### CODE REVIEW

**Review by anurag6713 — APPROVED** [review]
LGTM 🙂

**Review by hahmadia — APPROVED** [review]

File: components/admin_console/admin_definition.jsx [hunk 1]
```diff
@@ -5070,6 +5071,18 @@ const AdminDefinition = {
         sectionTitle: t('admin.sidebar.compliance'),
         sectionTitleDefault: 'Compliance',
         isHidden: it.not(it.userHasReadPermissionOnResource('compliance')),
+        global_policy_form: {
+            url: 'compliance/data_retention/global_policy',
+            isHidden: it.any(
+                it.not(it.licensedForFeature('DataRetention')),
+                it.configIsFalse('FeatureFlags', 'CustomDataRetentionEnabled'),
+            ),
+            isDisabled: it.not(it.userHasWritePermissionOnResource('compliance')),
```
[the remark this answers is not in the document]
Comment (BenCookie95) [reply components/admin_console/admin_definition.jsx:5080]: Good spot!

**Review by furqanmlk — APPROVED** [review]
LGTM

image

**Pull request merged**